tests/unit/scene/test_scene_dunder_methods.py

```diff
@@ -0,0 +1,65 @@
+import arcade
+import pytest
+
+
+def test_len():
+    scene = arcade.Scene()
+    scene.add_sprite_list("Player")
+
+    assert len(scene) == 1
+
+    scene.add_sprite_list("Walls")
+    scene.add_sprite_list("Coins")
+
+    assert len(scene) == 3
+
+def test_delitem():
+    scene = arcade.Scene()
+    scene.add_sprite_list("Walls")
+    scene.add_sprite_list("Player")
+    scene.add_sprite_list("Coins")
+    del scene["Player"]
+
+    with pytest.raises(KeyError):
+        scene["Player"]
+
+    del scene[scene._sprite_lists[0]]
+
+    with pytest.raises(KeyError):
+        scene["Walls"]
+
+    del scene[0]
+
+    with pytest.raises(KeyError):
+        scene["Coins"]
+
+def test_bool():
+    scene = arcade.Scene()
+    assert bool(scene) == False
```

Comment:
tl;dr although this works, the following is cleaner:
```suggestion
    assert not scene
```
`assert` handles non-`bool` data types based on whether they're [truthy or falsy](https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false).  For example, `assert 0` will fail because 0 is treated as falsy, while `assert 1` will succeed.